### PR TITLE
(Feat) YAML Interface to enhance the simulator usability

### PR DESCRIPTION
**Pull request by AlexCuadron**
## Overview

This pull request introduces a new interface for the simulation framework, focusing on the creation of a YAML configuration file and a Python script to set up and run simulations. Additionally, enhancements have been made to the client and replica modules to support the new interface.

### 1. `examples/simulator_example.yaml`

- **Purpose**: Provides a comprehensive example of how to configure the simulation using a YAML file.
- **Features**:
  - **Simulation Settings**: Configurable parameters such as output path, maximum ticks, step ticks, interactive mode, and progress bar display.
  - **Client Configurations**: Defines various client types, including `FixedTrafficClient`, `RandomChoiceWorkloadClient`, `RandomSendRequestClient`, and `DayAndNightClient`, each with specific parameters.
  - **Load Balancer Configuration**: Specifies the load balancing policy to be used.
  - **Replica Configurations**: Details the setup for replicas, including geographical location and accelerator type.

### 2. `simulator_interface.py`

- **Purpose**: Facilitates the setup and execution of simulations based on the YAML configuration.
- **Features**:
  - **YAML Loading**: Reads and parses the configuration from a specified YAML file.
  - **Simulation Setup**: Validates and extracts settings, clients, load balancer, and replicas from the configuration.
  - **Client and Replica Creation**: Utilizes factory methods to instantiate clients and replicas based on the configuration.
  - **Simulation Execution**: Calls the `_simulate_one` function to run the simulation with the configured parameters.

### 1. `client.py`

- **Enhancements**:
  - Introduced a registry for client types, allowing dynamic creation of client instances.
  - Added support for new client types with specific behaviors and configurations.
  - Improved the `make` method to facilitate client instantiation from type names.

### 2. `replica.py`

- **Enhancements**:
  - Implemented a registry for replica types, enabling flexible creation of replica instances.
  - Enhanced the `make` method to support replica instantiation from type names.
  - Added functionality to handle different types of replicas, such as `AcceleratorReplica`, with specific processing capabilities.

### CODE REVIEW

**Comment by AlexCuadron** [comment]
PTAL @cblmemo

**Review by cblmemo — COMMENTED** [review]
Thanks @AlexCuadron ! It is awesome 🚀 Mostly looks great to me. Left one discussion and it should be ready to go!

**Comment by AlexCuadron** [comment]
PTAL again! @cblmemo

File: client.py [hunk 1]
```diff
@@ -9,19 +9,110 @@
...
+            return client_cls(
+                traffics=traffic_objects,
+                location=location,
+                traffic_expired_time=traffic_expired_time,
+                period_tick=period_tick,
+            )
+        elif client_type == "random_choice":
+            if workload_candidates is None:
+                workload_candidates = []
+            return client_cls(
```
Comment (cblmemo) [inline client.py:88]: nit: could we have a `from_yaml_config` function in each subclass?

**Review by cblmemo — APPROVED** [review]
Thanks @AlexCuadron ! LGTM. Left a nit and after fixing that it should be ready to go!

Feel free to merge it once you fixed the nit ;)

**Pull request merged**